src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2110,12 +2110,18 @@ private unsafe ParsingError PrivateParseMinimal()
                             _flags |= Flags.UncPath;
                             idx = i;
                         }
+                        else if (!IsWindowsSystem && _syntax.InFact(UriSyntaxFlags.FileLikeUri) && pUriString[i - 1] == '/' && i - idx == 3)
```

Comment:
_syntax.InFact(UriSyntaxFlags.FileLikeUri) ensure the uri started with some scheme, so it will be either file:// or vsmacros://, and in the start of this function [here](https://github.com/Priya91/corefx-1/blob/f89975c699afe2d9eb1f81796b1208cdfdcb2159/src/System.Private.Uri/src/System/Uri.cs#L2060), we iterate through the string to pass over the scheme://, so i will not be 0. For readability, can put the i - idx == 3 check before this, which will guarantee i = idx + 3, and idx is non-negative, so i >= 3